SocketRocket/SRWebSocket.m

```diff
@@ -534,6 +534,14 @@ - (void)_initializeStreams;
     
     CFStreamCreatePairWithSocketToHost(NULL, (__bridge CFStringRef)host, port, &readStream, &writeStream);
     
+    //If voip is set in UIBackgroundMode, set apropriate properties on read and write stream
+    NSArray* backgroundModes = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"UIBackgroundModes"];
```

Comment:
Rather than detecting background modes manually and hard-coding iOS configuration strings into the project, it's probably better to use a Dependency Injection model and create a constructor that takes an array of service types to set for reading and writing (possibly with different parameters for the read and write stream). But I'll defer to @mikelikespie on the best way to get this functionality in here.